gql/parser.go

```diff
@@ -562,20 +580,182 @@ func parseArguments(l *lex.Lexer) (result []pair, rerr error) {
 	return result, nil
 }
 
+// String converts FilterTree to a string. Good for testing, debugging.
+func (t *FilterTree) String() string {
+	buf := bytes.NewBuffer(make([]byte, 0, 20))
+	t.stringHelper(buf)
+	return buf.String()
+}
+
+// stringHelper does simple DFS to convert FilterTree to string.
+func (t *FilterTree) stringHelper(buf *bytes.Buffer) {
+	if t == nil {
```

Comment:
A function on nil wouldn't crash in Go.. (recall the proverb: Make the zero value useful.)
It was there when I print out DebugStrings for all filters (even if they are nil).
It is no longer needed. Can I do an assert here to be sure?